st2api/tests/unit/controllers/v1/test_pack_configs.py

```diff
@@ -79,3 +78,16 @@ def test_put_pack_config(self):
                                           get_resp.json['values'], expect_errors=True)
         self.assertEqual(put_resp.status_int, 200)
         self.assertEqual(get_resp.json, put_resp_undo.json)
+
+    @mock.patch.object(PackConfigsController, '_dump_config_to_disk', mock.MagicMock())
+    def test_put_invalid_pack_config(self):
+        get_resp = self.app.get('/v1/configs/dummy_pack_11', params={'show_secrets': True},
+                                expect_errors=True)
```

Comment:
Hm... I wasn't expecting this to be in the DB. Didn't this config have the decrypt_kv as the config value?  Shouldn't we not allow the config to be loaded?

Reply:
Not sure what you mean, can you please clarify? I don't see where DB is coming from. I followed the example from the previous testcase, the self.app refers to the webtest app. Yes the config has the decrypt_kv as the config value, that is what we get in the values in the get_resp. Then we use that config and try to put it, which results in errors as expected.

Reply:
When you do a `get` on the webtest app, it is calling the st2api controller.  The API controller is reading from the database to get the config for `/v1/configs/dummy_pack_11`.  If there is validation error for the config because it is marked as secret and has the decrypt_kv in the config value, why is it in the DB in the first place where you can call `self.app.get('/v1/configs/dummy_pack_11')` to fetch it?

Reply:
@VineeshJain Have you verify if config marked secret with decrypt_kv in configured value will be loaded into the DB for production (at least w/ launchdev)?

Reply:
@VineeshJain Thanks for verifying that the bad config is not loaded into the DB.  When you get a chance, please close the loop here and for your edification, find out why the test here is loading the bad config into the DB.